mne/preprocessing/nirs/_temporal_derivative_distribution_repair.py

```diff
@@ -0,0 +1,143 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#          Frank Fishburn
+#
+# License: BSD (3-clause)
+
+
+import numpy as np
+
+from ... import pick_types
+from ...io import BaseRaw
+from ...utils import _validate_type
+
+
+def temporal_derivative_distribution_repair(raw):
+    r"""Apply temporal derivative distribution repair to data.
+
+    Applies temporal derivative distribution repair (TDDR) to data.
+    :footcite:`FishburnEtAl2019`. This approach removes baseline shift
+    and spike artifacts without the need for any user-supplied parameters.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data.
+    %(verbose)s
+
+    Returns
+    -------
+    raw : instance of Raw
+         Data with TDDR applied.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    raw = raw.copy().load_data()
+    _validate_type(raw, BaseRaw, 'raw')
+
+    if not len(pick_types(raw.info, fnirs='fnirs_od')):
+        raise RuntimeError('TDDR should be run on optical density data.')
+
+    raw._data = _TDDR(raw._data.T, raw.info['sfreq']).T
```

Comment:
This will operate on all of the data even if there are non-fNIRS channels. So probably needs `picks` incorporated

Reply:
Thanks. Good catch.